100/za/filter by id bug

Ziad-Agha opened the pull request:
- All the events in the manager dashboard are listed according to the manager's id.
- Managers only see their own events in their dashboard.
- Fixed bug where managers can only see their events in the event listing (from Home), now they can see all events.

Testing: 
- create a new Event as a manager and check whether is shows up in the dashboard regardless of what the contact email is.
- check that you can see all the events in the event listing.

Ziad-Agha committed:
added instruction to press on event to view analytics (#100)

Ziad-Agha committed:
event filtering for manager's dashboard now filters by ID (#100)

Ziad-Agha committed:
fixed a bug where the events for manager's are filtered to their events in the events listing (#100)

Ziad-Agha committed:
added a button to the event card isntead of a link (#100)

Ziad-Agha committed:
Simplify text for My Events overview

Removed the phrase about pressing on events to view analytics. (#100)

aminedakli reviewed (approved):
Tried it locally and everything works perfectly

Ziad-Agha committed:
Refactor attendance calculation and improve comments (#121) (#100)

Updated attendance calculation to query TicketPasses and added comments for clarity.

Ziad-Agha committed:
Update event analytics tests for clarity and accuracy (#100)

Refactor tests for event analytics to improve clarity and accuracy.

wiamekotbi reviewed (approved):
everything works as expected
